src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Avatar from '@material-ui/core/Avatar';
+import { Button } from '@material-ui/core';
+import './Dropdown.scss';
+
+interface Dropdownprops {
+  DropdownOpener: React.ReactNode;
+}
+
+export const Dropdown: React.FC<Dropdownprops> = ({ DropdownOpener }) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+  const box = useRef(null);
+
+  const closeDropdownOutside = (ref: React.MutableRefObject<HTMLDivElement | null>) => {
+    useEffect(() => {
+      const handleOutsideClick = (event: any) => {
+        if (ref.current && !ref.current.contains(event.target)) {
+          setIsDropdownOpen(false);
+          document.removeEventListener('click', handleOutsideClick);
+        }
+      };
+      if (isDropdownOpen) {
+        document.addEventListener('click', handleOutsideClick);
+      }
+    }, [isDropdownOpen, ref]);
+  };
+
+  closeDropdownOutside(box);
```

Comment:
you should use use effect for it, one time and you should return this function on unmount

return () => {
      clean event listener
    };